src/ima.py

```diff
@@ -846,7 +859,25 @@ def xbComputeLatency (self, mask):
 
                 elif (ex_op == 'mvm'):
                     mask_temp = self.de_xb_nma
-                    self.stage_latency[sId] = xbComputeLatency (self, mask_temp) # mask tells which of ip/op or both is occurring
+                    if (cfg.MVMU_ver == "Analog"):
+                        self.stage_latency[sId] = xbComputeLatency (self, mask_temp) # mask tells which of ip/op or both is occurring
+                    else:
+                        mvm_lat_temp = 0
+                        if (cfg.inference):
+                            for p in xrange(cfg.num_matrix):
+                                if self.de_xb_nma[p]:
+                                    xbar_inMem = self.xb_inMem_list[p]['f'].read_all ()
+                                    non_0_val = 0
+                                    for i in range(cfg.xbar_size):
+                                        if xbar_inMem[i] != '0':
+                                            non_0_val = non_0_val +1
+                                    print ("non_0_val", non_0_val)
+                                    nval_percent = int(non_0_val*100/128)
+                                    if (nval_percent%10!=0):
+                                        nval_percent = nval_percent + 10
+                                    mvm_lat_temp += param.Digital_xbar_lat_dict[cfg.MVMU_ver][str(cfg.xbar_size)][str(nval_percent)]
+                        self.stage_latency[sId] = mvm_lat_temp
+                    print("MVM Latency", self.stage_latency[sId])
```

Comment:
This should also be within the `xbComputeLatency` function (or have 2 functions `xbComputeLatency_analog/digital`)